bootstrap/012_task_log.sql

```diff
@@ -152,3 +160,52 @@ CREATE OR REPLACE VIEW ADMIN.TASK_LOG_HISTORY AS SELECT * FROM INTERNAL.TASK_LOG
 
 -- Remove view from old location
 DROP VIEW IF EXISTS REPORTING.TASK_LOG_HISTORY;
+
+CREATE OR REPLACE PROCEDURE INTERNAL.CLOSE_LOG_FOR_TASK(task_log_table text, query_history_table text, task_name text, object_name text)
+    RETURNS OBJECT
+    LANGUAGE SQL
+    Comment = 'Close any outstanding task rows that have not been updated since the given timestamp. If a query_id is present, the query_history row is set as the output'
+AS
+BEGIN
+    let outstanding_rows resultset := (select task_start, task_run_id, query_id from IDENTIFIER(:task_log_table) where success is null and task_name = :task_name and object_name = :object_name);
+    let cur cursor for outstanding_rows;
+    let rows_closed integer := 0;
+    for r in cur do
```

Comment:
sorry to be a pain bu tI am still striugglign as to why we need the for loop. I am really trying to get away from that.

Reply:
Not a pain, I didn't push my local changes after yesterday's distraction. Local change makes this whole proc an update stmt. Sorry for the confusion

Reply:
pushed the change